errors/errors.go

```diff
@@ -0,0 +1,99 @@
+package zqe
+
+import (
+	"bytes"
+	"fmt"
+	"runtime"
+)
+
+type Kind int
+
+const (
+	Other Kind = iota
+	Invalid
+	NotFound
+	Exists
+	Conflict
+)
+
+func (k Kind) String() string {
+	switch k {
+	case Other:
+		return "other error"
+	case Invalid:
+		return "invalid operation"
+	case NotFound:
+		return "item does not exist"
+	case Exists:
+		return "item already exists"
+	case Conflict:
+		return "conflict with pending operation"
+	}
+	return "unknown error kind"
+}
+
+type Error struct {
```

Comment:
Would you consider also adding a `Is()` method (https://godoc.org/errors#Is) to `Error`?

Reply:
Which I haven't thought about in depth, but maybe it would look `Kind`s first if available, and otherwise call the `Err.Is()`, if available. 

Or maybe instead clients who want this should just call `errors.Is(e.Err)` directly?

Reply:
I put in an `Is(target error) bool` method that only checks if the target is also a zqe.Error & if their Kind's are the same. I didn't go beyond that for now, as that seemed like it would fit a common pattern of:

```go
  err := doSomething()
  if errors.Is(err, zqe.E(zqe.Exists)) {
    // already there
  }
```

and if we wanted to check additional cases, you could add more errors.Is calls:

```go
  err := doSomething()  // returns zqe.E(zqe.NotExist, os.ErrClosed)
  if errors.Is(err, zqe.E(zqe.Exists)) {
    // ...
  } else if errors.Is(err, os.ErrClosed) {
   // this branch will executed
  }
```